src/Aspire.Hosting/Dcp/DcpDependencyCheck.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+using Aspire.Hosting.Dcp.Process;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Globalization;
+using Aspire.Hosting.Properties;
+
+namespace Aspire.Hosting.Dcp;
+
+internal sealed partial class DcpDependencyCheck
+{
+    // Docker goes to into resource saver mode after 5 minutes of not running a container (by default).
+    // While in this mode, the commands we use for the docker runtime checks can take quite some time
+    private const int WaitTimeForDcpInfoCommandInSeconds = 10;
+
+    [GeneratedRegex("[^\\d\\.].*$")]
+    private static partial Regex VersionRegex();
+
+    public static async Task EnsureDcpDependenciesAsync(string? dcpPath, bool hasContainers, string? containerRuntime, CancellationToken cancellationToken)
```

Comment:
Actually ... suggest a bit of a design change here to continue on the good work that @JamesNK was doing to make this more testable. We should define an IDcpDependencyCheck interface and have this class implement it, and make this method an instance method and injected it in via DI. That way when we write unit tests we can mock this test out.

Reply:
Added an IDcpDependencyCheckService interface and exposed as a DI service.